3rd_party/catch2/BUILD

```diff
@@ -0,0 +1,19 @@
+package(default_visibility = ["//visibility:public"])
+
+cc_library(
+    name = "catch2",
+    hdrs = [
+        "include/opentelemetry/3rd_party/catch.hpp",
```

Comment:
Why do you need this file?

Reply:
It might be useful for a test that wants to define its own main, but I removed for now so that there's only catch2_with_main